### PR TITLE
feat!: move to use microgen

### PR DESCRIPTION
**Pull request by arithmetic1728**
Please pay special attention to 
1. synth hack in `synth.py`
2. changes to handwritten file `google/cloud/monitoring_v3/query.py`

### CODE REVIEW

File: google/cloud/monitoring_v3/query.py [hunk 1]
```diff
@@ -460,12 +461,16 @@ def _build_query_params(self, headers_only=False, page_size=None):
...
             to a sensible value set by the API.
         """
-        params = {"name": self._project_path, "filter_": self.filter}
+        params = {"name": self._project_path, "filter": self.filter}
 
+        end_timestamp = timestamp.Timestamp()
+        end_timestamp.FromDatetime(self._end_time)
         params["interval"] = types.TimeInterval()
-        params["interval"].end_time.FromDatetime(self._end_time)
+        params["interval"].end_time = end_timestamp
```
Comment (software-dov) [inline google/cloud/monitoring_v3/query.py:469]: Nit: why not move "interval" into the dict definition?
```python
params = {..., "interval": types.TimeInterval()}
```

File: google/cloud/monitoring_v3/query.py [hunk 2]
```diff
@@ -481,9 +486,9 @@ def _build_query_params(self, headers_only=False, page_size=None):
             )
 
         if headers_only:
-            params["view"] = enums.ListTimeSeriesRequest.TimeSeriesView.HEADERS
+            params["view"] = monitoring_v3.ListTimeSeriesRequest.TimeSeriesView.HEADERS
         else:
-            params["view"] = enums.ListTimeSeriesRequest.TimeSeriesView.FULL
+            params["view"] = monitoring_v3.ListTimeSeriesRequest.TimeSeriesView.FULL
```
Comment (software-dov) [inline google/cloud/monitoring_v3/query.py:491]: Outrageous nit, feel free to ignore: while we're changing things, why not simplify this a little?
```python
tsv = monitoring_v3.ListTimeSeriesRequest.TimeSeriesView
params["view"] = tsv.HEADERS if headers_only else tsv.FULL
```

File: google/cloud/monitoring_v3/query.py (hunk 1 above)
Comment (software-dov) [inline google/cloud/monitoring_v3/query.py:469]: I don't think constructing the timestamp proto is necessary. Proto plus should be able to marshal from a `datetime`, like so:
```python
params["interval"].end_time = self._end_time
```

File: google/cloud/monitoring_v3/query.py [hunk 3]
```diff
@@ -460,12 +461,16 @@ def _build_query_params(self, headers_only=False, page_size=None):
...
+        end_timestamp = timestamp.Timestamp()
+        end_timestamp.FromDatetime(self._end_time)
         params["interval"] = types.TimeInterval()
-        params["interval"].end_time.FromDatetime(self._end_time)
+        params["interval"].end_time = end_timestamp
         if self._start_time:
-            params["interval"].start_time.FromDatetime(self._start_time)
+            start_timestamp = timestamp.Timestamp()
+            start_timestamp.FromDatetime(self._start_time)
+            params["interval"].start_time = start_timestamp
```
Comment (software-dov) [inline google/cloud/monitoring_v3/query.py:473]: Same as above: I think we can set `start_time` directly from a `datetime` object.
```python
params["interval"].start_time = self._start_time
```

File: google/cloud/monitoring_v3/query.py [hunk 4]
```diff
@@ -460,12 +461,16 @@ def _build_query_params(self, headers_only=False, page_size=None):
...
         """
-        params = {"name": self._project_path, "filter_": self.filter}
+        params = {"name": self._project_path, "filter": self.filter}
 
+        end_timestamp = timestamp.Timestamp()
+        end_timestamp.FromDatetime(self._end_time)
         params["interval"] = types.TimeInterval()
-        params["interval"].end_time.FromDatetime(self._end_time)
+        params["interval"].end_time = end_timestamp
         if self._start_time:
```
Comment (software-dov) [inline google/cloud/monitoring_v3/query.py:470]: We _may_ not need the conditional. If `self._start_time` is `None` there shouldn't be a problem setting it.

If this is true, we can simplify the interval construction a *lot*.
```python
params = {..., "interval": types.TimeInterval(start_time=self._start_time, end_time=self._end_time)}
```

File: synth.py [hunk 5]
```diff
@@ -34,96 +34,50 @@
     version="v3",
     bazel_target="//google/monitoring/v3:monitoring-v3-py",
     include_protos=True,
+    proto_output_path="google/cloud/monitoring_v3/proto"
```
Comment (software-dov) [inline synth.py:37]: I'm a little confused by this. We shouldn't be generating the regular python protos if we're using the microgenerator. Is there something else that needs them, or am I misinterpreting what this line does?

File: synth.py [hunk 6]
```diff
@@ -34,96 +34,50 @@
...
-    "from google.cloud.monitoring_v3.gapic import (\n"
-    "    notification_channel_service_client as notification_client)\n",
-)
-s.replace(
-    ["google/cloud/monitoring_v3/__init__.py"],
-    "notification_channel_service_client.NotificationChannelServiceClient",
-    "notification_client.NotificationChannelServiceClient",
-)
-
+# Synth hack due to googleapis and python-api-common-protos out of sync.
```
Comment (software-dov) [inline synth.py:44]: Is there an issue filed to bring them into sync?

**Review by software-dov — COMMENTED** [review]
Looks good, just some nits in query.py and synth.py

File: synth.py (hunk 5 above)
Comment (arithmetic1728) [reply synth.py:37]: It just copies over the *.proto files, it doesn't generate any python protos.

File: synth.py (hunk 6 above)
Comment (arithmetic1728) [reply synth.py:44]: https://github.com/googleapis/api-common-protos/issues/81

File: google/cloud/monitoring_v3/query.py (hunk 1 above)
Comment (arithmetic1728) [reply google/cloud/monitoring_v3/query.py]: done

File: google/cloud/monitoring_v3/query.py (hunk 2 above)
Comment (arithmetic1728) [reply google/cloud/monitoring_v3/query.py]: done

File: google/cloud/monitoring_v3/query.py (hunk 1 above)
Comment (arithmetic1728) [reply google/cloud/monitoring_v3/query.py]: done

File: google/cloud/monitoring_v3/query.py (hunk 4 above)
Comment (arithmetic1728) [reply google/cloud/monitoring_v3/query.py]: done

File: google/cloud/monitoring_v3/query.py (hunk 3 above)
Comment (arithmetic1728) [reply google/cloud/monitoring_v3/query.py]: done

File: synth.py (hunk 5 above)
Comment (software-dov) [reply synth.py:37]: Okay, good to know. Is this a backwards compatibility thing?

**Review by software-dov — APPROVED** [review]

File: synth.py (hunk 5 above)
Comment (arithmetic1728) [reply synth.py:37]: This is more like informational. The old client libs contains all the *.proto files so in the new one I just keep them.

File: google/cloud/monitoring_v3/query.py [hunk 7]
```diff
@@ -334,20 +334,20 @@ def align(self, per_series_aligner, seconds=0, minutes=0, hours=0):
 
         Example::
 
-            from google.cloud.monitoring import enums
+            import google.cloud.monitoring as monitoring
```
Comment (busunkim96) [inline google/cloud/monitoring_v3/query.py:337]:
```suggestion
            from google.cloud import monitoring
```
nit: This style is more common in samples and docs

File: google/cloud/monitoring_v3/query.py [hunk 8]
```diff
@@ -380,16 +380,16 @@ def reduce(self, cross_series_reducer, *group_by_fields):
         For example, you could request an aggregated time series for each
         combination of project and zone as follows::
 
-            from google.cloud.monitoring import enums
-            query = query.reduce(enums.Aggregation.Reducer.REDUCE_MEAN,
+            import google.cloud.monitoring as monitoring
```
Comment (busunkim96) [inline google/cloud/monitoring_v3/query.py:383]:
```suggestion
            from google.cloud import monitoring
```

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**